avm/ptn/azd/container-apps/main.bicep

```
@@ -0,0 +1,138 @@
+param internal bool = false
+
+@description('Conditional. IP range in CIDR notation that can be reserved for environment infrastructure IP addresses. It must not overlap with any other provided IP ranges and can only be used when the environment is deployed into a virtual network. If not provided, it will be set with a default value by the platform. Required if zoneRedundant is set to true  to make the resource WAF compliant.')
+param platformReservedCidr string = ''
+
+@description('Conditional. An IP address from the IP range defined by "platformReservedCidr" that will be reserved for the internal DNS server. It must not be the first address in the range and can only be used when the environment is deployed into a virtual network. If not provided, it will be set with a default value by the platform. Required if zoneRedundant is set to true to make the resource WAF compliant.')
+param platformReservedDnsIP string = ''
+
+@description('Conditional. Workload profiles configured for the Managed Environment. Required if zoneRedundant is set to true to make the resource WAF compliant.')
+param workloadProfiles array = []
+
+@description('Conditional. Name of the infrastructure resource group. If not provided, it will be set with a default value. Required if zoneRedundant is set to true to make the resource WAF compliant.')
+param infrastructureResourceGroupName string = take('ME_${containerAppsEnvironmentName}', 63)
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-containerapps.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module containerAppsEnvironment 'br/public:avm/res/app/managed-environment:0.7.0' = {
+  name: take('containerAppsEnvironment-${deployment().name}-deployment', 64)
+  params: {
+    name: containerAppsEnvironmentName
+    location: location
+    tags: tags
+    daprAIInstrumentationKey: daprAIInstrumentationKey
+    logAnalyticsWorkspaceResourceId: logAnalyticsWorkspaceResourceId
+    appInsightsConnectionString: appInsightsConnectionString
+    zoneRedundant: zoneRedundant
+    workloadProfiles: !empty(workloadProfiles) ? workloadProfiles : null
+    infrastructureResourceGroupName: infrastructureResourceGroupName
+    internal: internal
+    infrastructureSubnetId: !empty(infrastructureSubnetId) ? infrastructureSubnetId : null
```

Could be simlified to
```suggestion
    infrastructureSubnetId: infrastructureSubnetResourceId
```
if you make the parameter `nullable` using the `?`. 
Also note that `empty(null)` is `true`. So the conditions in the following lines will continue to work